myadmin-service-impls/student-services/src/main/java/za/ac/unisa/myadmin/student/services/impls/StudentServiceImpl.java

```diff
@@ -43,4 +46,100 @@ public String getSmartCardValue(Integer userId) throws OperationFailedException
 	public int updateSmartCardValue(String smartCard, Integer studentNumber) {
 		return studentRepository.updatesmartCardIssuedByStudentNumber(smartCard, studentNumber);
 	}
+
+	@Override
+	public List<StudentInfo> getStudentsBySurname(String surname) throws MissingParameterException, InvalidParameterException, OperationFailedException {
```

Comment:
These methods must throw a DoesNotExistsException if no student number was found.
If more than one result is found, we should carefully decide what to do. For me it is a valid response and the UI layer must handle that piece of logic if it wants to, because there could potentially be cases where you want to return all the student numbers found for the search criteria.